src/controller/artifact/processor/wasm/wasm.go

```diff
@@ -0,0 +1,130 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasm
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/goharbor/harbor/src/controller/artifact/processor"
+	"github.com/goharbor/harbor/src/controller/artifact/processor/base"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/artifact"
+	"github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// const definitions
+const (
+	// ArtifactTypeImage is the artifact type for image
+	ArtifactTypeWASM        = "WASM"
+	AdditionTypeBuildHistory = "BUILD_HISTORY"
+	mediaType        = "application/vnd.wasm.config.v1+json"
+)
+
+func init() {
+	pc := &manifestV2Processor{}
+	pc.ManifestProcessor = base.NewManifestProcessor()
+	mediaTypes := []string{
+		mediaType,
+	}
+	if err := processor.Register(pc, mediaTypes...); err != nil {
+		log.Errorf("failed to register processor for media type %v: %v", mediaTypes, err)
+		return
+	}
+}
+
+// manifestV2Processor processes image with OCI manifest and docker v2 manifest
+type manifestV2Processor struct {
+	*base.ManifestProcessor
+}
+
+func (m *manifestV2Processor) AbstractMetadata(ctx context.Context, art *artifact.Artifact, manifest_body []byte) error {
+
+	art.ExtraAttrs = map[string]interface{}{}
+	art.ExtraAttrs["ArtifactType"] = "WebAssembly"
+
+	manifest := &v1.Manifest{}
+	if err := json.Unmarshal(manifest_body, manifest); err != nil {
+		return err
+	}
+
+	if manifest.Annotations["module.wasm.image/variant"]=="compat" || manifest.Annotations["run.oci.handler"]=="wasm" {
+
+		// for annotation way
+		config := &v1.Image{}
+		if err := m.UnmarshalConfig(ctx, art.RepositoryName, manifest_body, config); err != nil {
+			return err
+		}
+		art.ExtraAttrs["manifest.config.mediaType"] = manifest.Config.MediaType
+		if art.ExtraAttrs == nil {
+			art.ExtraAttrs = map[string]interface{}{}
+		}
+		art.ExtraAttrs["created"] = config.Created
```

Comment:
Are those `created`, `architecture`, `os`, `config`, etc the attributes of WebAssembly? If they are not, I don't think the `AbstractMetadata()` is necessary

Reply:
yes, these flags are the attributes of WebAssembly l in annotation fashion wasm artifact.